### ISSUE TITLE
Improve the "dialog" when deploying to the server

### ISSUE THREAD

**Issue by odockal** [redhat-developer/vscode-adapters#342]
I was a bit confused that I actually could choose a folder with deployment (without need to name it xxx.war beforehand) when deploying to the wildfly server. I chose a folder and got hint "Do you want to edit optional deployment parameters?" which correctly guided me to choose a name for new deployment with war suffix, but still, it was not clear in the beginning.

**Comment by adietish** [comment]
steps:

1. ASSERT: have a wildfly server
1. EXEC: pick "Add deployment to server" in ctx menu
1. EXEC: choose a folder (exploded deployment) that has no extension in it's name
1. EXEC: when asked to provide Options (Yes/No?) say No

Result:
Deployment fails without giving any reason in the error dialog. See #330

Expected result:
Should have guidance that instructs user to provide a deployment type either in the folder-name or via the deployment Options

**Comment by robstryker** [comment]
Current code is as follows:

		for( int i = 0; i < supportedSuffix.length; i++ ) {
			if( hasSuffix(outputName, supportedSuffix[i]))
				return Status.OK_STATUS;
		}
		return new Status(IStatus.ERROR, Activator.BUNDLE_ID, 
				NLS.bind("Server {0} does not support deployment of resources without an approved suffix. " + 
		"Use the " + ServerManagementAPIConstants.DEPLOYMENT_OPTION_OUTPUT_NAME + " deployment option to override the output name.", server));


Doesn't this fix the issue?

**Comment by adietish** [comment]
@robstryker error is helpful, no doubt. But we should better inform the user upfront, before the error happens. I'm not fully happy with how it's currently to me things are still non-obvious, rely too much on hidden knowledge:
When adding a deployment how would you know that you have to say "Yes" if you add an exploded deployment:
![image](https://user-images.githubusercontent.com/25126/59352258-83ba3800-8d20-11e9-8ee5-a83dcf2f76e4.png)
It's only when providing the deployment name that you're told to provide the extension:
![image](https://user-images.githubusercontent.com/25126/59352402-d98ee000-8d20-11e9-894b-ac47b161cb54.png)

**Comment by robstryker** [comment]
I agree with you. I welcome specific suggestions on whether the language of the error message needs to change, or the API, or the workflow.

**Comment by robstryker** [comment]
> But we should also inform upfront, before the users possibly runs into the error.

How can we inform the user upfront? The rsp-server itself has no ability to inform anything since no request has been made of it yet (ie you didn't attempt to add a deployment yet).  

Furthermore, I don't believe the UI can possibly know what conditions could cause a deployment to fail without hard-coding specific situations for specific server types.  So I'm not sure how we can accomplish this goal.
